libbpf-tools/tcppktlat.bpf.c

```diff
@@ -70,26 +84,70 @@ static int handle_tcp_rcv_space_adjust(void *ctx, struct sock *sk)
 	if (delta_us < 0 || delta_us <= targ_min_us)
 		goto cleanup;
 
-	eventp = reserve_buf(sizeof(*eventp));
-	if (!eventp)
-		goto cleanup;
-
-	eventp->pid = pid;
-	eventp->tid = tid;
-	eventp->delta_us = delta_us;
-	eventp->sport = BPF_CORE_READ(inet, inet_sport);
-	eventp->dport = BPF_CORE_READ(sk, __sk_common.skc_dport);
-	bpf_get_current_comm(&eventp->comm, TASK_COMM_LEN);
-	family = BPF_CORE_READ(sk, __sk_common.skc_family);
-	if (family == AF_INET) {
-		eventp->saddr[0] = BPF_CORE_READ(sk, __sk_common.skc_rcv_saddr);
-		eventp->daddr[0] = BPF_CORE_READ(sk, __sk_common.skc_daddr);
-	} else { /* family == AF_INET6 */
-		BPF_CORE_READ_INTO(eventp->saddr, sk, __sk_common.skc_v6_rcv_saddr.in6_u.u6_addr32);
-		BPF_CORE_READ_INTO(eventp->daddr, sk, __sk_common.skc_v6_daddr.in6_u.u6_addr32);
+	if (targ_hist) {
```

Comment:
function is quite long now. Since the histogram and default modes have no overlap, How about separating them into functions?